.github/workflows/pr-auto-label.yml

```diff
@@ -0,0 +1,49 @@
+---
+name: PR-auto-labeler
+
+on:
+  pull_request_target:
+    types: [labeled, unlabeled, opened, edited, synchronize]
+
+permissions:
+  contents: read # the config file
+  pull-requests: write # for labeling pull requests (on: pull_request_target or on: pull_request)
+  statuses: write # to generate status
+  checks: write # to generate status
+
+jobs:
+  labeler:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check Labels
+        id: labeler
+        uses: jimschubert/labeler-action@v1
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  validate:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Fetch PR Body
+        id: fetch_body
+        run: |
+          body=$(curl -sSL -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" https://api.github.com/repos/${{ github.repository }}/pulls/${{ github.event.number }} | jq -r '.body')
+          echo "BODY<<EOF" >> $GITHUB_ENV
+          echo "$body" >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
+      - name: Check for labels in PR body
+        run: |
+          if [[ $(cat <<EOF 
+          ${{ env.BODY }} 
+          EOF) =~ /kind\ (fix|bugfix|feat|feature|enhancement|change|none|NONE|release-note-none|release-notes-none) ]] ; then
+            echo "PR body contains a valid type of change"
+          else
+            echo "PR body does not contain a valid type of change. Please refer to CONTRIBUTING.md for more information."
+            curl -sSL -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+              -X POST https://api.github.com/repos/${{ github.repository }}/issues/${{ github.event.number }}/comments \
+              -d '{"body": "PR body does not contain a valid type of change. Please refer to [CONTRIBUTING.md](https://github.com/prometheus/client_golang/blob/main/CONTRIBUTING.md#how-to-fill-the-pr-template) for more information."}'
+            exit 0
+          fi
```

Comment:
Oh wow, I'm impressed that you made this work like this :O 

But I'm not sure if we need to maintain something complex like this, GitHub Actions provide good support for triggering conditional steps.

Maybe we could simplify this a little bit?

Reply:
What immediately comes to mind is:

[`continue-on-error` for the labeler step](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error)

Another[ github-action](https://github.com/marketplace/actions/comment-pull-request) that is capable of commenting on PRs, and a [conditional on this step that will only execute if the previous one failed](https://docs.github.com/en/actions/learn-github-actions/contexts#:~:text=steps.%3Cstep_id%3E.conclusion,is%20success.)

Reply:
The [labeler action](https://github.com/jimschubert/labeler-action) we are currently using never fails; it always reports success, even if no labels are added. This makes it unsuitable for conditional steps.

I searched for alternative labeler actions that return a failure status but none offer the same level of [regex configuration](https://github.com/jimschubert/labeler-action?tab=readme-ov-file#full-schema) as our current one.

Since `labeler action` can't detect the absence of labels, I had to use GitHub API to fetch the PR description and post a comment 🙂

Reply:
@ArthurSens Thanks for recommending this [GitHub Action](https://github.com/marketplace/actions/label-checker-for-pull-requests)! Our workflow is much cleaner now :)

A comment is added if a label is missing, and it is automatically deleted when the label is added.